File: Py/g.func.py
from typing import Callable

def calculate_price(base_price: float, discount: Callable[[float], float] = lambda x: x) -> float:
    # Pure function to calculate final price with an optional discount function.
    return discount(base_price)

# Usage
no_discount = calculate_price(100.0)  # Identity function (no change)
fixed_discount = calculate_price(100.0, lambda x: x * 0.9)  # 10% off
custom_discount = calculate_price(100.0, lambda x: x - 20 if x > 50 else x)  # $20 off if > $50

print(f"No discount: ${no_discount:.2f}")      # $100.00
print(f"10% off: ${fixed_discount:.2f}")       # $90.00
print(f"Custom discount: ${custom_discount:.2f}")  # $80.00

File: C#/G.Func.cs
public class PriceCalculator
{
    public float CalculatePrice(float basePrice, Func<float, float>? discount = null)
    {
        // Default to identity function if no discount provided
        discount ??= x => x;
        return discount(basePrice);
    }

    void Main()
    {
        float noDiscount = CalculatePrice(100.0f);
        float fixedDiscount = CalculatePrice(100.0f, x => x * 0.9f);  // 10% off
        float customDiscount = CalculatePrice(100.0f, x => x > 50 ? x - 20 : x);  // $20 off if > $50

        Console.WriteLine($"No discount: ${noDiscount:F2}");      // $100.00
        Console.WriteLine($"10% off: ${fixedDiscount:F2}");       // $90.00
        Console.WriteLine($"Custom discount: ${customDiscount:F2}");  // $80.00
    }   
}